DuggaSys/tests/showDuggaservice_test.php

```diff
@@ -0,0 +1,160 @@
+        'query-before-test-3' => "SELECT MAX(lid) FROM listentries",
+        'query-after-test-1' => "DELETE FROM listentries WHERE cid = 9999;",
+        'query-after-test-2' => "DELETE FROM course WHERE cid = 9999;",
+        'query-variables' => "moment",
+        'service' => 'localhost/LenaSYS/DuggaSys/showDuggaservice.php',
+        'service-data' => serialize(array( // Data that service needs to execute function
+            'moment' => '<!query-before-test3!> <[0][listentries]>'
+        )),
+        'filter-output' => serialize(array( // Filter what output to use in assert test, use none to use all ouput from service
+            'none'
+        )),
+    ),
+    'super-view data on quizname' => array(
+        'expected-output' => '{"debug":"[Guest] Missing hash\/password\/variant!","param":"{}","answer":"UNK","danswer":"UNK","score":0,"highscoremode":"","grade":"UNK","submitted":"","marked":"","deadline":"UNK","release":"UNK","files":[],"userfeedback":"UNK","feedbackquestion":"UNK","variant":"UNK","ishashindb":false,"variantsize":"UNK","variantvalue":"UNK","password":"UNK","hashvariant":"UNK","isFileSubmitted":"UNK","isTeacher":0,"variants":[],"duggaTitle":"UNK","hash":"UNK","hashpwd":"UNK","opt":"UNK","link":"UNK"}',
+        'query-before-test-1' => "INSERT INTO course(cid, creator) VALUES (9999, 1);",
+        'query-before-test-2' => "INSERT INTO listentries(cid,entryname,link,kind,pos,creator,visible,vers,gradesystem,highscoremode,feedbackenabled,feedbackquestion) VALUES (9999, 'Inserttobedeleted', 'UNK', 4, 12, 2, 1, 1337, 1, 1, 0, 'UNK');",
+        'query-before-test-3' => "SELECT MAX(lid) FROM listentries",
+        'query-after-test-1' => "DELETE FROM listentries WHERE cid = 9999;",
+        'query-after-test-2' => "DELETE FROM course WHERE cid = 9999;",
+        'query-variables' => "moment",
+        'service' => 'localhost/LenaSYS/DuggaSys/showDuggaservice.php',
+        'service-data' => serialize(array( // Data that service needs to execute function
+            'moment' => '<!query-before-test3!> <[0][listentries]>'
+        )),
+        'filter-output' => serialize(array( // Filter what output to use in assert test, use none to use all ouput from service
+            'none'
+        )),
+    ),
+    'view data on userAnswer' => array(
+        'expected-output' => '{"debug":"[Guest] Missing hash\/password\/variant!","param":"{}","answer":"UNK","danswer":"UNK","score":0,"highscoremode":"","grade":"UNK","submitted":"","marked":"","deadline":"UNK","release":"UNK","files":[],"userfeedback":"UNK","feedbackquestion":"UNK","variant":"UNK","ishashindb":false,"variantsize":"UNK","variantvalue":"UNK","password":"UNK","hashvariant":"UNK","isFileSubmitted":"UNK","isTeacher":0,"variants":[],"duggaTitle":"UNK","hash":"UNK","hashpwd":"UNK","opt":"UNK","link":"UNK"}',
```

Comment:
query-before-test needed for INSERT INTO course(cid, creator) VALUES (9999, 1);

Reply:
query-before-test2 needed for INSERT INTO listentries(cid,entryname,link,kind,pos,creator,visible,vers,gradesystem,highscoremode,feedbackenabled,feedbackquestion) VALUES (9999, "Inserttobedeleted", "UNK", 4, 12, 2, 1, 1337, 1, 1, 0, "UNK");

Reply:
another query-before-test3 needed for SELECT MAX(lid) FROM listentries, this value has to be saved in query-variables as moment

Reply:
another query-before-test4 needed for INSERT INTO userAnswer (cid,quiz,moment,vers,variant,hash,password, useranswer) VALUES(9999, 52432, 1, ? , 3, “ghj1ghj2” , “asddasdd”,NULL)

NOTE: this query is completely wrong in the test-case file, make sure these values are correct.